### DISCUSSION TITLE
[General] era's blocks confirmations, estimateBlockTime, reorgPeriod?

### DISCUSSION THREAD

**Discussion by nooma-42** [zkSync-Community-Hub/zksync-developers#258] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

what's era's blocks confirmations, estimateBlockTime, reorgPeriod?

### What is the specific issue or error you're encountering?

not specified in doc, dc bot can't answer

### Can you share the error messages or logs you're receiving, if any?

not specified in doc, dc bot can't answer

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

ask dc bot, consult the doc

**Comment by Romsters** [comment, marked as answer]
Hi @NOOMA-42 , if I understand your question correctly transaction confirmations is a number of blocks added after the block which includes the transaction. In terms of finality there are a couple of stages a transaction goes though until it is considered finalised. More details [here](https://era.zksync.io/docs/reference/concepts/finality.html#finality-on-zksync-era), for transactions statuses/stages: [here](https://era.zksync.io/docs/reference/concepts/transactions.html#transaction-statuses).

Regarding `estimateBlockTime`, blocks are normally generated every 1 second, means there is a new block generated every second.

On `reorgPeriod`, in rare cases, where we cannot prove the block's correctness i.e. we mistakenly committed an unprovable block or someone committed a malicious block, revertBlock reverts all changes that came with that block and prevents such data from being included in subsequent blocks. Only non-finalized blocks can be reverted. Once a block is finalized, it cannot be reverted.

**Reply by savasadar** [reply]
Are you sure about blocks are generated every 1 second? Because When I check from explorer, It seems generating too fast. I did not find any technical document about block time interval.

**Reply by savasadar** [reply]
I checked from block countdown API and always api and remaining time in seconds value is same. That is, 1 block is created every second. Thanks

https://block-explorer-api.mainnet.zksync.io/docs#/Block%20API/ApiController_getBlockCountdown

```
{
  "status": "1",
  "message": "OK",
  "result": {
    "CurrentBlock": "28263063",
    "CountdownBlock": "28282992",
    "RemainingBlock": "19929",
    "EstimateTimeInSec": "19929"
  }
}
```

**Comment by belanoparo** [comment]
Block confirmations, estimate block time, and reorg period are essential parameters in blockchain networks, including zkSync. However, these specific parameters are not explicitly documented for zkSync in the available resources. To resolve this, it's recommended to directly inquire with the zkSync development team or community channels for accurate information regarding these parameters.

**Comment by procebetjumorales** [comment]
Era's block confirmations, estimateBlockTime, and reorgPeriod are important parameters related to the blockchain network's stability, security, and performance. Unfortunately, these specific details are not provided in the documentation or available through the DC bot. For accurate information, it's recommended to consult the official era documentation or reach out to the era development team directly.

**Comment by dgeyston** [comment]
Era's blocks confirmations, estimateBlockTime, and reorgPeriod are essential parameters related to the blockchain consensus mechanism, block confirmation time estimation, and reorganization period. These details are crucial for understanding the behavior and reliability of the era blockchain network.

**Comment by mariyokaveen** [comment]
The era blockchain's block confirmations, estimateBlockTime, and reorgPeriod parameters are not specified in the documentation or accessible via the Discord bot.

**Comment by bryanbrega** [comment]
The functionality being asked about includes era's blocks confirmations, estimateBlockTime, and reorgPeriod. However, these details are not specified in the documentation and the DC bot is unable to provide an answer. No error messages or logs are provided, and there have been no recent changes to the contract. There are no external libraries or contracts involved, and the relevant portions of the contract code are not provided. The user has attempted to consult the documentation and ask the DC bot for help.
